execution/evm/vm.go

```diff
@@ -134,12 +134,9 @@ func (vm *VM) fireCallEvent(eventSink EventSink, callType exec.CallType, errProv
 	errSink.PushError(eventErr)
 }
 
-// CONTRACT state is aware of caller and callee, so we can just mutate them.
-// CONTRACT code and input are not mutated.
-// CONTRACT returned 'ret' is a new compact slice.
-// value: To be transferred from caller to callee. Refunded upon errors.CodedError.
-// gas:   Available gas. No refunds for gas.
-// code: May be nil, since the CALL opcode may be used to send value from contracts to accounts
+// Initiate an EVM call against the provided state pushing events to eventSink. code should contain the EVM bytecode,
+// input the CallData (readable by CALLDATALOAD), value the amount of native token to transfer with the call
```

Comment:
This is not entirely correct. In the deploy code path the constructor arguments are concatenated with the code, and not stored in the calldataload. This is just another EVM weirdism.